astropy/cosmology/io/row.py

```diff
@@ -59,8 +62,13 @@ def from_row(row, *, move_to_meta=False, cosmology=None):
         FlatLambdaCDM(name="Planck18", H0=67.66 km / (Mpc s), Om0=0.30966,
                       Tcmb0=2.7255 K, Neff=3.046, m_nu=[0. 0. 0.06] eV, Ob0=0.04897)
     """
+    inv_rename = {v: k for k, v in rename.items()} if rename is not None else {}
+    kname = inv_rename.get("name", "name")
+    kmeta = inv_rename.get("meta", "meta")
+    kcosmo = inv_rename.get("cosmology", "cosmology")
+
     # special values
-    name = row["name"] if "name" in row.columns else None  # get name from column
+    name = row[kname] if kname in row.columns else None  # get name from column
```

Comment:
https://github.com/astropy/astropy/commit/d87cde4144b926eb2ca1831bbb18483cd9ad2b2b is merged now, so after rebasing it would be possible to write
```suggestion
    name = row.get(kname)
```